src/net_processing.cpp

```diff
@@ -1723,6 +1730,12 @@ void PeerManagerImpl::BlockConnected(const std::shared_ptr<const CBlock>& pblock
             m_txrequest.ForgetTxHash(ptx->GetWitnessHash());
         }
     }
+    auto stalling_timeout = m_block_stalling_timeout.load();
+    // In case the dynamic timeout was doubled once or more, reduce it slowly back to its default value
```

Comment:
In the normal case, the timeout is already at the default, so probably good to optimise for that:

```c++
if (stalling_timeout > DEFAULT_BLOCK_STALLING_TIMEOUT) {
```

Reply:
Done, makes sense to me.